src/Uno.UI.Tests/Windows_UI_Xaml/Given_Implicit_Style.cs

```diff
@@ -23,5 +32,273 @@ public void When_Implicit_Style_In_Application_Merged()
 
 			Assert.AreEqual(Colors.LightGoldenrodYellow, (control.TestCheckBox.Foreground as SolidColorBrush).Color);
 		}
+
+		[TestMethod]
+		public void When_Implicit_Style_In_Visual_Tree_Local_Type()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			var strBefore = control.InlineTemplateControl.MyStringDP;
+
+			app.HostView.Children.Add(control);
+
+			var strAfter = control.InlineTemplateControl.MyStringDP;
+
+			Assert.AreEqual("AppLevelImplicit", strBefore);
+			Assert.AreEqual("InnerTreeImplicit", strAfter);
+		}
+
+		[TestMethod]
+		public void When_Implicit_Style_In_Visual_Tree_Framework_Type()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			var tagBefore = control.TestRadioButton.Tag;
+
+			app.HostView.Children.Add(control);
+
+			var tagAfter = control.TestRadioButton.Tag;
+
+			Assert.AreEqual("AppLevelImplicit", tagBefore);
+			Assert.AreEqual("InnerTreeImplicit", tagAfter);
+		}
+
+		[TestMethod]
+		public void When_Newly_Created_Custom_Type()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var styledControl = new StylesTestControl();
+
+			var styledButton = new StylesTestButton();
+
+			Assert.AreEqual("FromDefault", styledControl.Rugosity);
+			Assert.AreEqual(new Thickness(), styledButton.Padding);
+
+			app.HostView.Children.Add(styledControl);
+			app.HostView.Children.Add(styledButton);
+
+			Assert.AreEqual("TopImplicitStyleValue", styledControl.Rugosity);
+			Assert.AreEqual(DefaultButtonPadding, styledButton.Padding);
+		}
+
+		[TestMethod]
+		public void When_Framework_Type_Default_Style()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var button = new Button();
+
+			app.HostView.Children.Add(button);
+
+			Assert.IsNull(button.Style); //Default style is not actually set to the Style property
+			Assert.AreEqual(DefaultButtonPadding, button.Padding);
+		}
+
+		[TestMethod]
+		public void When_Newly_Created_Framework_Type_Set_Style()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var button = new Button();
+
+			var style = app.Resources["ExplicitButtonStyle"] as Style;
+			button.Style = style;
+			Assert.AreEqual(new Thickness(), button.Padding); //Default style not applied yet, we haven't been added to the visual tree.
+		}
+
+		[TestMethod]
+		public void When_Newly_Created_Framework_Type_ApplyTemplate()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'app'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)